src/polycubed/src/port.cpp

```diff
@@ -287,5 +297,47 @@ void Port::unconnect(PeerIface &p1, PeerIface &p2) {
   p2.set_peer_iface(nullptr);
 }
 
+void Port::subscribe_event(const std::string &param_name,
+      std::function<void(const std::string, const std::string)> callback) {
+  if (param_name == "MAC_ADDRESS" || param_name == "IP_ADDRESS") {
+    // Add event to the list
+    subscription_list.emplace(param_name, callback);
+
+    // Check if the port already has a peer
+    ExtIface* extiface = dynamic_cast<ExtIface*>(get_peer_iface());
+    if (extiface)
+      extiface->subscribe_event(param_name, callback);
+
+  } else {
+    logger->error("subscribeEvent: Parameter name {0} not available", param_name);
+    throw std::runtime_error("subscribeEvent: Parameter name " + param_name + " not available");
```

Comment:
this is bad a practice to log and throw an error, just throw the error and the one calling this function should perform the logging of the exception.